src/globals.d

```diff
@@ -14,26 +14,50 @@ import ddmd.root.array;
 import ddmd.root.filename;
 import ddmd.root.outbuffer;
 
-template xversion(string s)
-{
-    enum xversion = mixin(`{ version (` ~ s ~ `) return true; else return false; }`)();
-}
-
 private string stripRight(string s)
 {
     while (s.length && (s[$ - 1] == ' ' || s[$ - 1] == '\n' || s[$ - 1] == '\r'))
         s = s[0 .. $ - 1];
     return s;
 }
 
-enum IN_GCC     = xversion!`IN_GCC`;
+immutable bool TARGET_LINUX, TARGET_OSX, TARGET_FREEBSD, TARGET_OPENBSD,
+               TARGET_SOLARIS, TARGET_WINDOS, TARGET_POSIX;
 
-enum TARGET_LINUX   = xversion!`linux`;
-enum TARGET_OSX     = xversion!`OSX`;
-enum TARGET_FREEBSD = xversion!`FreeBSD`;
-enum TARGET_OPENBSD = xversion!`OpenBSD`;
-enum TARGET_SOLARIS = xversion!`Solaris`;
-enum TARGET_WINDOS  = xversion!`Windows`;
+shared static this()
```

Comment:
Well, could you explain how you plan on getting rid of selecting file name extensions like .o, .obj, .so, .a, etc. without using these `TARGET_*` runtime variables?  If you mean you simply will use differently named runtime variables, these can always be renamed later, so I'd call this a step in that direction.  It doesn't go all the way there simply because I don't know what scheme you have in mind and it's outside the scope of this PR, which is just to get rid of `static if` where possible.